query-service-impl/src/main/java/org/hypertrace/core/query/service/pinot/converters/PinotFunctionConverter.java

```diff
@@ -34,24 +40,77 @@ public PinotFunctionConverter() {
   }
 
   public String convert(
-      Function function, java.util.function.Function<Expression, String> argumentConverter) {
+      ExecutionContext executionContext,
+      Function function,
+      java.util.function.Function<Expression, String> argumentConverter) {
     switch (function.getFunctionName().toUpperCase()) {
       case QUERY_FUNCTION_COUNT:
         return this.convertCount();
       case QUERY_FUNCTION_PERCENTILE:
         return this.functionToString(this.toPinotPercentile(function), argumentConverter);
       case QUERY_FUNCTION_CONCAT:
         return this.functionToString(this.toPinotConcat(function), argumentConverter);
+      case QUERY_FUNCTION_AVG_RATE:
+        return this.functionToStringForAvgRate(function, argumentConverter, executionContext);
       default:
         // TODO remove once pinot-specific logic removed from gateway - this normalization reverts
         // that logic
         if (this.isHardcodedPercentile(function)) {
-          return this.convert(this.normalizeHardcodedPercentile(function), argumentConverter);
+          return this.convert(
+              executionContext, this.normalizeHardcodedPercentile(function), argumentConverter);
         }
         return this.functionToString(function, argumentConverter);
     }
   }
 
+  private String functionToStringForAvgRate(
+      Function function,
+      java.util.function.Function<Expression, String> argumentConverter,
+      ExecutionContext executionContext) {
+    function = updateFunctionForAvgRate(function, executionContext);
+    String argumentString =
+        function.getArgumentsList().stream()
+            .map(argumentConverter::apply)
+            .collect(Collectors.joining(","));
+
+    return "SUM(DIV(" + argumentString + "))";
+  }
+
+  private Function updateFunctionForAvgRate(Function function, ExecutionContext executionContext) {
+
+    Expression columnName = function.getArgumentsList().get(0);
+    Expression literal = function.getArgumentsList().get(1);
+
+    Builder builder = function.toBuilder();
+    builder.clearArguments();
+    builder.addArguments(columnName);
+    builder.addArguments(getUpdatedLiteral(columnName, literal, executionContext));
+
+    return builder.build();
+  }
+
+  private Expression getUpdatedLiteral(
+      Expression column, Expression literal, ExecutionContext executionContext) {
+
+    String columnName = column.getColumnIdentifier().getColumnName().split("[.]")[0];
+    long divisorInSeconds = literal.getLiteral().getValue().getLong();
+    double timeRangeInSeconds =
+        executionContext
+            .getTimeSeriesColumn(columnName)
+            .orElse(executionContext.getTimeFilterDuration(columnName));
+
+    return Expression.newBuilder()
+        .setLiteral(
+            LiteralConstant.newBuilder()
+                .setValue(
+                    Value.newBuilder()
+                        .setDouble(timeRangeInSeconds / divisorInSeconds)
```

Comment:
nit: can we rename this vars? 
`divisorInSeconds -> rateInervalInSeconds`
`timeRangeInSeconds -> aggrgateInervalInSeconds`